src/sframe/sframe.hpp

```diff
@@ -360,7 +360,9 @@ class sframe : public swriter_base<sframe_output_iterator> {
    * column id is out of range.
    */
   inline std::string column_name(size_t i) const {
-    ASSERT_MSG(i < index_info.column_names.size(), "Index out of range!");
+    if(i >= index_info.column_names.size()) {
+      log_and_throw("Index out of range!");
```

Comment:
Since these error messages get propagated all the way to the user, and they may not have context for how it's called, we should add  a bit more description.  Maybe "Column index out of range."